### PR TITLE
fix(ui) : add default value to host groups and service groups

### PR DESCRIPTION
**Pull request by AssanDiomande**
## Description

**Fixes** MON-11771

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [x] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

Check if the value in the pie chart is "100% undertemined" for a host groups / service groups not binded to a host

### CODE REVIEW

File: www/include/reporting/dashboard/DB-Func.php [hunk 1]
```diff
@@ -158,6 +158,12 @@ function getLogInDbForHostGroup($hostgroup_id, $start_date, $end_date, $reportTi
 
     $hosts_id = $centreon->user->access->getHostHostGroupAclConf($hostgroup_id, 'broker');
     if (count($hosts_id) == 0) {
+        $hostgroupStats["average"]["UP_MP"] = 0;
```
Comment (kduret) [inline www/include/reporting/dashboard/DB-Func.php:161]: I think labels could be added here : https://github.com/centreon/centreon/blob/ce48b838ee0c213a01e63c193b40d265bd9eee83/www/include/reporting/dashboard/common-Func.php#L116
just be sure there are no side effects

File: www/include/reporting/dashboard/common-Func.php [hunk 2]
```diff
@@ -116,7 +116,7 @@ function getPeriodToReport(?string $alternate = null): array
 function getHostStatsValueName()
 {
     return (
-        array(
+        [
```
Comment (kduret) [inline www/include/reporting/dashboard/common-Func.php:119]: missing `TOTAL_ALERTS`

File: www/include/reporting/dashboard/common-Func.php [hunk 3]
```diff
@@ -141,7 +144,7 @@ function getHostStatsValueName()
 function getServicesStatsValueName()
 {
     return (
-        array(
+        [
```
Comment (kduret) [inline www/include/reporting/dashboard/common-Func.php:147]: missing `TOTAL_ALERTS`

File: www/include/reporting/dashboard/DB-Func.php [hunk 4]
```diff
@@ -158,6 +158,12 @@ function getLogInDbForHostGroup($hostgroup_id, $start_date, $end_date, $reportTi
...
+        $hostgroupStats["average"]["DOWN_MP"] = 0;
+        $hostgroupStats["average"]["UNREACHABLE_MP"] = 0;
+        $hostgroupStats["average"]["UNDETERMINED_TP"] = 100;
+        $hostgroupStats["average"]["MAINTENANCE_TP"] = 0;
+        $hostgroupStats["average"]["TOTAL_ALERTS"] = 0;
```
Comment (kduret) [inline www/include/reporting/dashboard/DB-Func.php:166]:
```suggestion
        $hostgroupStats["average"]["UNDETERMINED_TP"] = 100;
```

File: www/include/reporting/dashboard/DB-Func.php [hunk 5]
```diff
@@ -543,6 +549,16 @@ function getLogInDbForServicesGroup($servicegroupId, $startDate, $endDate, $repo
     $count = 0;
     $services = getServiceGroupActivateServices($servicegroupId);
 
+    if (empty($services)) {
+        $serviceGroupStats["average"]["UNDETERMINED_TP"] = 100;
+        $serviceGroupStats["average"]["TOTAL_ALERTS"] = 0;
+        $serviceGroupStats["average"]["OK_MP"] = 0;
+        $serviceGroupStats["average"]["WARNING_MP"] = 0;
+        $serviceGroupStats["average"]["CRITICAL_MP"] = 0;
+        $serviceGroupStats["average"]["UNKNOWN_MP"] = 0;
```
Comment (kduret) [inline www/include/reporting/dashboard/DB-Func.php:558]:
```suggestion
        $serviceGroupStats["average"]["UNDETERMINED_TP"] = 100;
```

**Review by kduret — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Pull request merged**